test/metabase/driver_test.clj

```diff
@@ -79,6 +80,39 @@
                          :field-definitions [{:field-name "foo", :base-type :type/Text}]
                          :rows              [["bar"]]}]}))
 
+(deftest can-connect-with-destroy-db-test
+  (testing "driver/can-connect? should fail or throw after destroying a database"
+    (mt/test-drivers (->> (mt/normal-drivers)
+                          ;; athena is a special case because connections aren't made with a single database,
+                          ;; but to an S3 bucket that may contain many databases
+                          (remove #{:athena}))
+      (let [database-name (mt/random-name)
+            dbdef         (basic-db-definition database-name)]
+        (mt/dataset dbdef
+          (let [db (mt/db)
+                details (tx/dbdef->connection-details driver/*driver* :db dbdef)]
+            (testing "can-connect? should return true before deleting the database"
+              (true? (binding [h2/*allow-testing-h2-connections* true]
```

Comment:
Is this intentionally just `(true? ..)` rather than `(is (true? ...))`? The same with `false?` down there.

Reply:
Ha, I'm an idiot. I was surprised this passed on the first go without any changes. I'll fix this up